cmake/modules/OmrUtility.cmake

```diff
@@ -139,43 +139,66 @@ function(omr_count_true out)
 	set("${out}" "${result}" PARENT_SCOPE)
 endfunction(omr_count_true)
 
-# return the path of the target ${tgt} in VAR
-# (only executable and shared library targets are currently supported)
-function(omr_get_target_path VAR tgt)
+# omr_genex_property_chain(<output_var> <target> [<property> ...] <fallback>)
+#   Create a long generator expression chain which will results in the first
+#   <property> value (of <target>) which evaluates to non-false value, or
+#   <fallback> if they all evaluate to false
+#   Psuedo code for the resulting generator expression (Note: evaluated at generate time)
+#   for (i := 0 to N )
+#      x := $<TARGET_PROPERTY:target,property[i]>
+#      if(x)
+#          return x
+#      else
+#          continue
+#   return fallback;
+function(omr_genex_property_chain output_var target )
+	omr_assert(TEST ARGC GREATER "2" MESSAGE "No fallback case provided")
+
+	list(REVERSE ARGN)
+	# start off the generator expression with the fallback case, which is now
+	# at the front of the list
+	list(GET ARGN 0 out)
+	list(REMOVE_AT ARGN 0)
+
+	foreach(prop_name IN LISTS ARGN)
+
+		set(prop_genex "$<TARGET_PROPERTY:${target},${prop_name}>")
+		set(prop_bool "$<BOOL:${prop_genex}>")
+		set(out "$<${prop_bool}:${prop_genex}>$<$<NOT:${prop_bool}>:${out}>")
+	endforeach()
+	set(${output_var} "${out}" PARENT_SCOPE)
+endfunction()
+
+# omr_get_target_output_info(<target> <output_dir_var> <output_name_var>)
+#   Returns a pair of generator expressions.
+#   <output_dir_var> receives a generator expression which yields the output directory or <target>
+#   <output_name_var> receives a generator expression which yields the output name of <target>
+#   with the file suffix omitted.
+#   Note: currenly only executable and shared library targets are supported
+function(omr_get_target_output_info tgt output_dir_var output_name_var)
 	get_target_property(target_type ${tgt} TYPE)
 	if(target_type STREQUAL "EXECUTABLE")
-		get_target_property(target_directory ${tgt} RUNTIME_OUTPUT_DIRECTORY)
-		get_target_property(target_name      ${tgt} RUNTIME_OUTPUT_NAME)
-		if(NOT target_name)
-			set(target_name "${tgt}${CMAKE_EXECUTABLE_SUFFIX}")
-		endif()
+		set(search_prefixes "RUNTIME_" "")
 	elseif(target_type STREQUAL "SHARED_LIBRARY")
-		get_target_property(target_directory ${tgt} LIBRARY_OUTPUT_DIRECTORY)
-		get_target_property(target_name      ${tgt} LIBRARY_OUTPUT_NAME)
-		if(NOT target_name)
-			set(target_name "lib${tgt}${CMAKE_SHARED_LIBRARY_SUFFIX}")
-		endif()
+		set(search_prefixes "LIBRARY_" "")
 	else()
 		message(FATAL_ERROR "omr_get_target_path: ${tgt} type is ${target_type}")
```

Comment:
This message should be updated to reflect the updated function name (`omr_get_target_output_info`).